### PR TITLE
[otp_ctrl] Transition to D1

### PR DESCRIPTION
**Pull request by msfschaffner**
This PR is dependent on #3720 (cleanup) and #3721 (instantiation in top level). Only the last commit in this PR is relevant.

### CODE REVIEW

**Comment by msfschaffner** [comment]
Rebased on top of master - OTP is ready to transition into D1 now.

**Comment by msfschaffner** [comment]
CC @mateenfaisal

File: hw/ip/otp_ctrl/doc/checklist.md [hunk 1]
```diff
@@ -10,21 +10,21 @@ stage, and updated as needed. Once done, please remove this comment before check
 This checklist is for [Hardware Stage]({{< relref "/doc/project/development_stages.md" >}}) transitions for the [OTP_CTRL peripheral.]({{< relref "hw/ip/otp_ctrl/doc" >}})
 All checklist items refer to the content in the [Checklist.]({{< relref "/doc/project/checklist.md" >}})
 
-## Design Checklist
+<%text>## Design Checklist</%text>
```
Comment (cindychip) [inline hw/ip/otp_ctrl/doc/checklist.md:13]: Do you mind also changing this from the template file?

File: hw/ip/otp_ctrl/doc/checklist.md [hunk 2]
```diff
@@ -10,21 +10,21 @@ stage, and updated as needed. Once done, please remove this comment before check
...
+Documentation | [SPEC_COMPLETE][]     | Done        | [OTP_CTRL Design Spec]({{<relref "hw/ip/otp_ctrl/doc" >}})
+Documentation | [CSR_DEFINED][]       | Done        |
+RTL           | [CLKRST_CONNECTED][]  | Done        |
+RTL           | [IP_TOP][]            | Done        |
+RTL           | [IP_INSTANTIABLE][]   | Done        |
+RTL           | [MEM_INSTANCED_80][]  | Done        |
+RTL           | [FUNC_IMPLEMENTED][]  | Done        |
+RTL           | [ASSERT_KNOWN_ADDED][]| Done        |
+Code Quality  | [LINT_SETUP][]        | Done        |
 Review        | Reviewer(s)           | Not Started |
```
Comment (cindychip) [inline hw/ip/otp_ctrl/doc/checklist.md:28]: Do we need a reviewer for D1?

File: hw/ip/otp_ctrl/doc/checklist.md (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/otp_ctrl/doc/checklist.md]: right, I forgot to delete that from the template. thanks.

File: hw/ip/otp_ctrl/doc/checklist.md (hunk 2 above)
Comment (msfschaffner) [reply hw/ip/otp_ctrl/doc/checklist.md:28]: @sjgitty I thought this is not needed anymore, right? should we delete these last two lines from the template?

Comment (sjgitty) [reply hw/ip/otp_ctrl/doc/checklist.md:28]: > @sjgitty I thought this is not needed anymore, right? should we delete these last two lines from the template?

Yes, I think it should be removed from the template. Feel free to remove now, or if you keep it we'll remove from all checklists when we do the template PR.

**Review by cindychip — APPROVED** [review]
LGTM!

File: hw/ip/otp_ctrl/doc/checklist.md (hunk 2 above)
Comment (msfschaffner) [reply hw/ip/otp_ctrl/doc/checklist.md:28]: ok let's do that as a separate cleanup in order to not "pollute" the D0 -> D1 transition PR of OTP.

**Pull request merged**